configbp/config.go

```diff
@@ -0,0 +1,69 @@
+// Package configbp parses configurations per the baseplate spec.
+package configbp
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"gopkg.in/yaml.v2"
+
+	"github.com/reddit/baseplate.go/log"
+)
+
+// ExpandConfig performs baseplate-conformant configuration expansion.
+//
+// Currently, this includes expanding $FOO and ${FOO} with their respective environment variables.
+func ExpandConfig(rawConfig string) string {
+	expanded := os.ExpandEnv(rawConfig)
+	if expanded != rawConfig {
+		log.Debugf("Expanded configuration:\n%s", expanded)
```

Comment:
🔕 in general I'm against using direct log functions in a library code, but consider the use case of this library this is probably ok.

Reply:
I'd kinda like to get my ctxlog package into baseplate at some point, and that could potentially let us do it a bit more cleanly, but yeah.  Not my favorite but these kinds of things have been life-savers in the past.

Reply:
We are using ` fmt.Fprintln(os.Stderr` in some other [places](https://github.com/reddit/baseplate.go/blob/master/runtimebp/config.go#L40) though I find debug level preferable as it is easier to suppress noisy messages when you are aware of them.

Reply:
@konradreiche there are two reasons we used `fmt.Fprintln(os.Stderr)` there:

1. `runtimebp` is a standalone library, it does not need to be tied with our logging library (people can choose to use it without using any other baseplate.go libraries)
2. at the time `log.Debug/Info/Error/etc.` goes nowhere before we initialized them (that was fixed in efa21f216085c6239e587f65563def72fc51bd88)

Reply:
I've changed around how the logging is done -- please take a look and let me know what you think